bugfixes: user account update, create user

tuurma opened the pull request:
solves eXist-db/dashboard#93 and eXist-db/dashboard#76

adamretter commented:
@tuurma I took a look over your PR, thanks for that :-)

It looks good, but as far as I can see, it doesn't solve eXist-db/dashboard#93 as claimed. Thoughts?

joewiz commented on modules/userManager.xqm:
```diff
@@ -167,7 +167,7 @@ declare function usermanager:update-user($user-name as xs:string, $user-json as
 
                 (: if a password is provided, we always change the password, as we dont know what the original password was :)
-                for $group in secman:get-user-groups($user) return secman:remove-group-member($group, $user),
+                for $group in secman:get-user-groups($user)[. != sm:get-user-primary-group($user)] return secman:remove-group-member($group, $user),
```
It might be best to stick with the unusual `secman` prefix defined in the prolog, or otherwise switch the whole module to the standard `sm`.

tuurma replied:
yeah, it does get on my nerves and, as demonstrated, is easy to make a silly mistake

joewiz commented:
On Monday's Community Call @duncdrum suggested that fixing this issue would be a good opportunity to review https://github.com/eXist-db/usermanager/pull/25 (a PR which moves the Cypress integration tests into this repository), update it to reflect the recent changes to the login behavior, and to add a test showing that the fix to this issue has the expected results.

tuurma commented:
@adamretter correct, didn't read the exist-db/dashboard#93 title properly: there were two parts to that issue
* updating the user in any way was failing (now fixed)
* resulting 500 error (now replaced with a similar problem with a login redirect)

currently working on proper fix for the second part

tuurma commented:
@joewiz Thanks, I do realise that it would be ideal to add relevant tests asap
